### PR TITLE
Implement Lasso Container

### PR DESCRIPTION
**Pull request by Jiarui-Cao**
### Description
Implemented class for Lasso Container and tested sample models

### Testing
Integration testing in `pysrc/test/integration/test_lasso_container.py` using some non-trivial data stored in `pysrc/test/integration/resources`

### CODE REVIEW

File: src/pysrc/model/lasso_container.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+from abc import ABC
+from pathlib import Path
+from pysrc.util.types import Asset
+from pysrc.util.lasso_utils import asset_to_model_filename, search_model_in_directory
+from pysrc.util.pickle_utils import load_pickle
+
+
+class LassoContainer(ABC):
+    """
```
Comment (echavemann) [inline src/pysrc/model/lasso_container.py:9]: no docstrings lmao

File: src/pysrc/test/integration/model/test_lasso_container.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+from pysrc.model.lasso_container import LassoContainer
+import pytest
+
+resource_path = Path(__file__).parent.parent.joinpath("resources")
+
+
+def test_lasso_container_failed_initialization() -> None:
+    error_msg = "LassoContainer initialized without an instantiated model for one ore more assets"
+
+    # Asset model not instantiated
```
Comment (echavemann) [inline src/pysrc/test/integration/model/test_lasso_container.py:15]: del thse comments, we should be able to figure out

File: src/pysrc/test/integration/model/test_lasso_container.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import numpy as np
+from pathlib import Path
+from pysrc.util.types import Asset
+from pysrc.util.enum_conversions import enum_to_string
+from pysrc.util.pickle_utils import load_pickle, store_pickle
+from pysrc.model.lasso_container import LassoContainer
+import pytest
+
+resource_path = Path(__file__).parent.parent.joinpath("resources")
```
Comment (echavemann) [inline src/pysrc/test/integration/model/test_lasso_container.py:9]: there should be a test helper function for this already

File: src/pysrc/test/integration/model/test_lasso_container.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    with pytest.raises(AssertionError) as excinfo:
+        LassoContainer(resource_path, [Asset.BTC, Asset.ETH, Asset.DOGE, Asset.ADA])
+    assert str(excinfo.value) == error_msg
+
+
+def test_lasso_container_successful_initialization() -> None:
+    LassoContainer(resource_path, [])
+    LassoContainer(resource_path, [Asset.BTC])
+    LassoContainer(resource_path, [Asset.ETH, Asset.DOGE])
+    LassoContainer(resource_path, [Asset.BTC, Asset.ETH, Asset.DOGE])
```
Comment (echavemann) [inline src/pysrc/test/integration/model/test_lasso_container.py:30]: i know annoying, but can we test one with all of them?

Comment (echavemann) [reply src/pysrc/test/integration/model/test_lasso_container.py:30]: actually nvm

File: src/pysrc/test/lasso_data_helpers.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            directory_path.joinpath(asset_to_model_filename(asset))
+        )
+        score = lasso_model.score(X_test, y_test)
+        print(f"{asset} model R-squared score: {score}")
+
+    print(type(X_test))
+    print(X_test.shape, y_test.shape, test_data.shape)
+
+
+# if __name__ == '__main__':
```
Comment (echavemann) [inline src/pysrc/test/lasso_data_helpers.py:96]: delete

File: src/pysrc/test/lasso_data_helpers.py
Comment (echavemann) [inline src/pysrc/test/lasso_data_helpers.py:1]: delete the prints

File: src/pysrc/util/pickle_utils.py
Comment (echavemann) [inline src/pysrc/util/pickle_utils.py:1]: tests for these?

File: src/pysrc/model/lasso_container.py (hunk 1 above)
Comment (Jiarui-Cao) [reply src/pysrc/model/lasso_container.py]: Removed. Why tho?

File: src/pysrc/test/integration/model/test_lasso_container.py (hunk 2 above)
Comment (Jiarui-Cao) [reply src/pysrc/test/integration/model/test_lasso_container.py]: del'ed

File: src/pysrc/test/integration/model/test_lasso_container.py (hunk 3 above)
Comment (Jiarui-Cao) [reply src/pysrc/test/integration/model/test_lasso_container.py]: used `ger_resource_path()`

File: src/pysrc/test/lasso_data_helpers.py (hunk 5 above)
Comment (Jiarui-Cao) [reply src/pysrc/test/lasso_data_helpers.py]: del'ed

File: src/pysrc/test/lasso_data_helpers.py
Comment (Jiarui-Cao) [reply src/pysrc/test/lasso_data_helpers.py:1]: del'ed

File: src/pysrc/util/pickle_utils.py
Comment (Jiarui-Cao) [reply src/pysrc/util/pickle_utils.py:1]: added integration test for `store_pickle()` and `load_pickle()`

File: src/pysrc/model/lasso_container.py (hunk 1 above)
Comment (echavemann) [reply src/pysrc/model/lasso_container.py]: code speaks for itself

Comment (echavemann) [reply src/pysrc/model/lasso_container.py]: if you need comments its either a very exceptional case (unlikely) or your variable names and typenames are shit (more liikley)

File: src/pysrc/test/integration/util/test_pickle_utils.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import numpy as np
+from pathlib import Path
+from pysrc.util.pickle_utils import store_pickle, load_pickle
+from pysrc.test.helpers import get_resources_path
+
+resource_path = get_resources_path(str(Path(__file__).parent))
+
+
+def test_store_and_load_picle() -> None:
```
Comment (echavemann) [inline src/pysrc/test/integration/util/test_pickle_utils.py:9]: id seperate these into different tests, fix spelling, and probably make the resources path so yo can just do `get_resources_path(__file__)`

File: src/pysrc/model/lasso_container.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
+from abc import ABC
+from pathlib import Path
+from pysrc.util.types import Asset
+from pysrc.util.lasso_utils import asset_to_model_filename, search_model_in_directory
+from pysrc.util.pickle_utils import load_pickle
+
+
+class LassoContainer(ABC):
```
Comment (echavemann) [inline src/pysrc/model/lasso_container.py:8]: wait why is this an ABC

Comment (Jiarui-Cao) [reply src/pysrc/model/lasso_container.py:8]: ugh my bad removed

File: src/pysrc/test/integration/util/test_pickle_utils.py (hunk 6 above)
Comment (Jiarui-Cao) [reply src/pysrc/test/integration/util/test_pickle_utils.py]: Yea I refactored the resource directories. Separated the tests too

**Review by echavemann — APPROVED** [review]

**Pull request merged**